src/main/webapp/assets/js/view/application-add-wizard.js

```diff
@@ -680,10 +680,12 @@ define([
                 this.$('.required-config-loading').hide()
                 if (catalogEntryItem!=null && catalogEntryItem.config!=null) {
                     var that = this
-                    _.each(catalogEntryItem.config, function (cfg) {
-                        if (cfg.priority !== undefined) {
+                    _.chain(catalogEntryItem.config).sortBy(function (cfg){
+                        return !isNaN(cfg.priority) ? cfg.priority : 99999;
+                    }).each(function (cfg) {
+                        if (cfg.pinned === true || (cfg.contraints.indexOf('required') > -1 && typeof cfg.defaultValue === 'string' && cfg.defaultValue === '')) {
```

Comment:
Agree, that's my point.
`defaultValue == null` on the server side is serialized as missing (`undefined`) on the wire.
Which means that `cfg.contraints.indexOf('required') > -1 && typeof cfg.defaultValue === 'string' && cfg.defaultValue === ''` will be `false` for `undefined`. So it will not be shown, even if required.